src/python/bot/fuzzers/afl/launcher.py

```diff
@@ -1251,6 +1252,29 @@ def associate_features_with_file(self, features, path):
       self._associate_feature_with_element(feature, element)
 
 
+def _verify_system_config():
+  """Verifies system settings required for AFL."""
+
+  def _check_core_pattern_file():
+    """Verifies that core pattern file content is set to 'core'."""
+    if not os.path.exists(constants.CORE_PATTERN_FILE_PATH):
+      return False
+
+    return open(constants.CORE_PATTERN_FILE_PATH).read().strip() == 'core'
+
+  if _check_core_pattern_file():
+    return
+
+  return_code = subprocess.call(
+      'timeout 2s sudo bash -c "echo core > {path}"'.format(
```

Comment:
why do we need the timeout?

Reply:
If we don't have sudoer access, this allows to bailout from the subprocess command (subprocess.call timeout arg only in py3) and then properly exit on log fatal, rather than leading to infinite hang (since we would be stuck to wait on user to enter sudo pass).

Reply:
how about using -n ?

     -n, --non-interactive
                 Avoid prompting the user for input of any kind.  If a password is required for the command to run, sudo will display an
                 error message and exit.

Reply:
Done!